walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,176 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
 
+// Class containing the query method to find available time ranges for a given meeting request.
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    // Find optional time ranges for the meeting.
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
```

Comment:
SMALL PRs!!

As far as I understand, you implemented the "Optional Coding Challenge", which is great, but it screams an incremental change.

I can't possibly review this code, it'd be better if you start with the simple problem and then extend it

Reply:
This part was actually mandatory and not the optional challenge. I felt like it was a big PR but in the tutorial it said we should  open a PR when we're done...
Anyway - I'm sorry. I'll try to see how to divide it into 2 smaller PRs.